lib/onebox/engine/amazon_onebox.rb

```diff
@@ -30,7 +32,11 @@ def match
       def image
         case
         when raw.css("#main-image").any?
-          raw.css("#main-image").first["src"]
+          ::JSON.parse(
+            raw.css("#main-image").first
+              .attributes["data-a-dynamic-image"]
+              .value
+          ).keys.first
```

Comment:
```html
<script type="text/javascript">
if(typeof encodedImgSrc != 'undefined' && encodedImgSrc != "" && document.getElementById("main-image")!=null){document.getElementById("main-image").src=encodedImgSrc;};setCSMReq('af');setCSMReq('af');
addlongPoleTag('af','awudplphtml');
</script>
```

```html
<img alt="" src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7" onload="" class="a-dynamic-image" id="main-image" data-a-dynamic-image='{"http://ecx.images-amazon.com/images/I/51opYcR6kVL._SY400_.jpg":[200,166]}'>
```

There has been some changes with Amazon that sets the `src` of the image using Javascript.